### PR TITLE
Upgrades ojdbc to the latest version of 23 as of this writing

### PR DESCRIPTION
**Pull request by ljnelson**
See also: #9351.

### CODE REVIEW

**Comment by ljnelson** [comment]
Looks like the UCP changed in some non-backwards compatible way between 21.x and 23.x; investigating

**Comment by ljnelson** [comment]
A test involving previously successful injection of a `PoolDataSource` with `fred` as a `serviceName` is now failing with, ultimately:
```
Caused by: java.lang.AssertionError: service mismatch: this.service=fred, smServiceName=
	at oracle.ucp.common.CoreConnectionImpl.<init>(CoreConnectionImpl.java:124)
```
Not sure what `smServiceName` is; need more research on the actual UCP classes' bytecode.

**Comment by ljnelson** [comment]
The latest commit on this PR deliberately removes all special-cased property handling we were doing before, letting the UCP do the validation work instead. This is both more elegant (the special-casing was never going to scale, and this very PR got burned by it in a prior commit) and required (the upgrade of UCP from 21.x to 23.x introduced assertions in the UCP codebase that were not there previously and which were preventing the upgrade). Open to all comments on this situation.

**Review by barchetta — COMMENTED** [review]
Overall this looks good. Are you aware of any customer impact of this in the area of compatibility? If so, can you summarize in the PR Description?

**Comment by ljnelson** [comment]
From some debugging, it looks like the native image test is not failing due to any change made by this PR, but by the omission of a `--enable-url-protocols=http` native image option. I'm not sure why this PR in particular uncovered this omission or why this test has not failed before. The error message does not show up in the test run but does appear in the downloadable test archive, which includes standard output.

**Comment by ljnelson** [comment]
Adding `--enable-url-protocols=http` to `tests/integration/packaging/mp-2/src/main/resources/META-INF/native-image/io.helidon.tests.integration.packaging/helidon-tests-integration-packaging-mp2/native-image.properties` and re-running the test (via `main` again) now yields:
```
Caused by: org.graalvm.nativeimage.MissingReflectionRegistrationError: The program tried to reflectively invoke method public java.lang.String(java.lang.String) without it being registered for runtime reflection. Add public java.lang.String(java.lang.String) to the reflection metadata to solve this problem. See https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection for help.
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.forQueriedOnlyExecutable(MissingReflectionRegistrationUtils.java:72)
	at java.base@21.0.3/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:74)
	at java.base@21.0.3/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base@21.0.3/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at org.glassfish.jersey.internal.inject.ParamConverters$StringConstructor$1._fromString(ParamConverters.java:143)
	at org.glassfish.jersey.internal.inject.ParamConverters$AbstractStringReader.fromString(ParamConverters.java:92)
	at org.glassfish.jersey.server.internal.inject.AbstractParamValueExtractor.convert(AbstractParamValueExtractor.java:116)
	at org.glassfish.jersey.server.internal.inject.AbstractParamValueExtractor.fromString(AbstractParamValueExtractor.java:107)
	at org.glassfish.jersey.server.internal.inject.SingleValueExtractor.extract(SingleValueExtractor.java:61)
	at org.glassfish.jersey.server.internal.inject.PathParamValueParamProvider$PathParamValueProvider.apply(PathParamValueParamProvider.java:92)
	at org.glassfish.jersey.server.internal.inject.PathParamValueParamProvider$PathParamValueProvider.apply(PathParamValueParamProvider.java:79)
	at org.glassfish.jersey.server.spi.internal.ParamValueFactoryWithSource.apply(ParamValueFactoryWithSource.java:50)
	at org.glassfish.jersey.server.spi.internal.ParameterValueHelper.getParameterValues(ParameterValueHelper.java:64)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$AbstractMethodParamInvoker.getParamValues(JavaResourceMethodDispatcherProvider.java:109)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
```

**Comment by ljnelson** [comment]
I've pushed [dbf4198](https://github.com/helidon-io/helidon/pull/9441/commits/dbf41985b5761eae9e0d1b442e75ab47ffc23d87) following the instructions in the `mp-2` test failure. This change is technically out of scope for this PR but I'd like other Graal-related errors to start showing up.

**Comment by ljnelson** [comment]
Good (I guess); with the Graal-suggested command line option of `--enable-url-protocols=http` now the pipeline (not just my machine) is reporting:
```
org.graalvm.nativeimage.MissingReflectionRegistrationError: The program tried to reflectively invoke method public java.lang.String(java.lang.String) without it being registered for runtime reflection. Add public java.lang.String(java.lang.String) to the reflection metadata to solve this problem. See https://www.graalvm.org/latest/reference-manual/native-image/metadata/#reflection for help.
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.MissingReflectionRegistrationUtils.forQueriedOnlyExecutable(MissingReflectionRegistrationUtils.java:72)
	at java.base@21.0.3/java.lang.reflect.Constructor.acquireConstructorAccessor(Constructor.java:74)
	at java.base@21.0.3/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base@21.0.3/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at org.glassfish.jersey.internal.inject.ParamConverters$StringConstructor$1._fromString(ParamConverters.java:143)
	at org.glassfish.jersey.internal.inject.ParamConverters$AbstractStringReader.fromString(ParamConverters.java:92)
	at org.glassfish.jersey.server.internal.inject.AbstractParamValueExtractor.convert(AbstractParamValueExtractor.java:116)
	at org.glassfish.jersey.server.internal.inject.AbstractParamValueExtractor.fromString(AbstractParamValueExtractor.java:107)
	at org.glassfish.jersey.server.internal.inject.SingleValueExtractor.extract(SingleValueExtractor.java:61)
	at org.glassfish.jersey.server.internal.inject.PathParamValueParamProvider$PathParamValueProvider.apply(PathParamValueParamProvider.java:92)
	at org.glassfish.jersey.server.internal.inject.PathParamValueParamProvider$PathParamValueProvider.apply(PathParamValueParamProvider.java:79)
	at org.glassfish.jersey.server.spi.internal.ParamValueFactoryWithSource.apply(ParamValueFactoryWithSource.java:50)
	at org.glassfish.jersey.server.spi.internal.ParameterValueHelper.getParameterValues(ParameterValueHelper.java:64)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$AbstractMethodParamInvoker.getParamValues(JavaResourceMethodDispatcherProvider.java:109)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:274)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:266)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:253)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:696)
	at io.helidon.microprofile.server.JaxRsService.doHandle(JaxRsService.java:228)
	at io.helidon.microprofile.server.JaxRsService.lambda$handle$2(JaxRsService.java:185)
	at io.helidon.common.context.Contexts.runInContext(Contexts.java:117)
	at io.helidon.microprofile.server.JaxRsService.handle(JaxRsService.java:185)
	at io.helidon.webserver.http.HttpRoutingImpl$RoutingExecutor.doRoute(HttpRoutingImpl.java:165)
	at io.helidon.webserver.http.HttpRoutingImpl$RoutingExecutor.call(HttpRoutingImpl.java:124)
	at io.helidon.webserver.http.HttpRoutingImpl$RoutingExecutor.call(HttpRoutingImpl.java:102)
	at io.helidon.webserver.http.ErrorHandlers.runWithErrorHandling(ErrorHandlers.java:76)
	at io.helidon.webserver.http.Filters$FilterChainImpl.proceed(Filters.java:121)
	at io.helidon.common.context.Contexts.runInContext(Contexts.java:117)
	at io.helidon.webserver.context.ContextRoutingFeature.filter(ContextRoutingFeature.java:50)
	at io.helidon.webserver.http.Filters$FilterChainImpl.proceed(Filters.java:119)
	at io.helidon.webserver.http.Filters.executeFilters(Filters.java:87)
	at io.helidon.webserver.http.Filters.lambda$filter$0(Filters.java:83)
	at io.helidon.webserver.http.ErrorHandlers.runWithErrorHandling(ErrorHandlers.java:76)
	at io.helidon.webserver.http.Filters.filter(Filters.java:83)
	at io.helidon.webserver.http.HttpRoutingImpl.route(HttpRoutingImpl.java:73)
	at io.helidon.webserver.http1.Http1Connection.route(Http1Connection.java:544)
	at io.helidon.webserver.http1.Http1Connection.handle(Http1Connection.java:223)
	at io.helidon.webserver.ConnectionHandler.run(ConnectionHandler.java:169)
	at io.helidon.common.task.InterruptableTask.call(InterruptableTask.java:47)
	at io.helidon.webserver.ThreadPerTaskExecutor$ThreadBoundFuture.run(ThreadPerTaskExecutor.java:239)
	at java.base@21.0.3/java.lang.Thread.runWith(Thread.java:1596)
	at java.base@21.0.3/java.lang.VirtualThread.run(VirtualThread.java:309)
	at java.base@21.0.3/java.lang.VirtualThread$VThreadContinuation$1.run(VirtualThread.java:190)
```

**Comment by ljnelson** [comment]
Following the instructions output by GraalVM and with @tvallin's verification and help I've added `reflect-config.json` to this native image test. The PR itself does not require it, nor is the error related to the PR at all. I am still not sure how the `mp-2` test has ever passed if this reflection configuration information is suddenly necessary now, but wasn't before.

**Comment by ljnelson** [comment]
Partial mystery solved: the Oracle JDBC `ojdbc11` artifact, at version 21.15.0.0, has the following in its `native-image.properties` file it ships with:
```
  -H:EnableURLProtocols=http \
  -H:EnableURLProtocols=https
```
These lines are not present in the `native-image.properties` file shipped with `ojdbc11` version 23.6.0.24.10 (nor, I would think, should they be). So that explains that issue.

**Comment by ljnelson** [comment]
Following #9510, I have removed `reflect-config.json` from the `mp-2` test and have instead introduced a provisional `reflect-config.json` in `jersey/server`.

**Review by barchetta — DISMISSED** [review]

**Comment by ljnelson** [comment]
Rebased to accommodate Netty and OCI upgrades in `dependencies/pom.xml`.

**Review by barchetta — DISMISSED** [review]

**Review by barchetta — APPROVED** [review]
